device/sycl/src/utils/global_index.hpp

```diff
@@ -0,0 +1,36 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/device/global_index.hpp"
+
+// SYCL include(s).
+#include <sycl/sycl.hpp>
+
+namespace traccc::sycl::details {
+
+/// Function creating a global index in a 1D SYCL kernel
+inline device::global_index_t global_index(const ::sycl::nd_item<1>& item) {
+
+    return static_cast<device::global_index_t>(item.get_global_linear_id());
+}
+
+/// Function creating a global index in a 2D SYCL kernel
+inline device::global_index_t global_index(const ::sycl::nd_item<2>& item) {
+
+    return static_cast<device::global_index_t>(item.get_global_linear_id());
+}
+
+/// Function creating a global index in a 3D SYCL kernel
+inline device::global_index_t global_index(const ::sycl::nd_item<3>& item) {
+
+    return static_cast<device::global_index_t>(item.get_global_linear_id());
+}
```

Comment:
This seems very silly to me; we're replacing a SYCL native call to `get_global_linear_id` by a custom function which simply wraps it for no good reason _and_ also unsafely narrows the value to a 32-bit integer... This is just unnecessary complexity plus unnecessary code fragility; suggest we keep it as it was.

Reply:
Or, indeed, we use the thread identifier types, which are intended for exactly this purpose.

Reply:
The main reason I introduced these helper functions for SYCL is to do the static casts in as few places as possible.

Let's stop worrying about not fitting into 32 bits with our thread count! I've never seen any NVIDIA engineer advise against using `unsigned int` with this. (I believe the hardware has much lower limits than 2^32 threads...)

SYCL rightfully used `std::size_t` in its functions. That's what I would've done as well. But as long as we need to use `unsigned int` for indices "for atomic reasons", we're stuck with that type... 😦

Reply:
FYI the maximum grid size in CUDA is $2^{57}$ threads, so there certainly is reason to be worried.

Also, thinking of it, the maximum number of threads in the $x$-axis is $2^{41}$ so that also presents some issues with the unsigned integer type. It's $2^{26}$ threads in the $y$ and $z$ axes, so no problems there.

But okay, fine, we can do it this way.